resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -76,13 +76,12 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   private def setUpExecutorConfigMap(driverPod: Option[Pod]): Unit = {
     val configMapName = KubernetesClientUtils.configMapNameExecutor
-    val resolvedExecutorProperties =
-      Map(KUBERNETES_NAMESPACE.key -> conf.get(KUBERNETES_NAMESPACE))
     val confFilesMap = KubernetesClientUtils
```

Comment:
Yep, It's reasonable in here, because namespace is in `configmap.metadata`, we couldn't get any namespace from confFilesMap, and also should not force put the namespaces into confFilesMap to avoid occupy any `configmap.data` key(such like the `namespace` in here).

Reply:
Hi, Yikun. Thanks for you commnet. I think `configmap.namespace` should keep up with `spark.kubernetes.namespace`, which means the namespace that will be used for running the driver and executor pods. And, executor configmap never set namespace, so I put the namespace into confFilesMap.

Reply:
Thanks for clarify, but the Kubernetes allow any key as configmap key (even if the key name is 'namespace'), we should't break any behavior due to spark implemention or workaround. 


So I think any key put in configmap should be covert to the real configmap.key rather than do some hack and special process to occupy any user potential input.


If we just need to set executor separately, I guess what we real needed is introducing a new conf like spark.k8s.executor.configmap.namespace otherwise back to spark.k8s.namespace directly.


And I also didn't see any problem if we just use conf.namespace like this PR done. Could you explain some If you have any other concern?

Reply:
I understood but I don't agree with that. First, driver confileMaps sets namespace in the same ways, aiming to keep confMaps up with sparkConf. Second, `spark.k8s.executor.configmap.namespace` is not necessary. Driver/executor pod and related resource should keep in same namespace.

Reply:
> driver confileMaps sets namespace in the same ways

Do you mean [this](https://github.com/apache/spark/blob/c4a9772f741836cdb399e35a45b3b5df48d9eea6/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientUtils.scala#L92) which was introduced in https://github.com/apache/spark/pull/34983 ? Actually this is the reason of why we need this PR (regression for driver side) and where `confFileMap` mixup problem happened.

> aiming to keep confMaps up with sparkConf.

Aggree, this is right, and this PR also can keep this regression, you can see @chia7712 already added the test for driver and executor and also avoid `confFileMap` mixup problem. 

> Second, spark.k8s.executor.configmap.namespace is not necessary.

I said **if we need**, I updated it in my last comments to avoid misleading. And also it decribed how we set  namespace according conf (even if the potential separate configure case in future) rather a indirect step confFileMap.

Of course, this is just my prefer, so I give my +1 on this PR.

Reply:
Of course, this is just my prefer, so I give my +1 on this PR. If the confiFilemap way can **solve the key mixup problem** and **also have regresson on driver side**, I'd also okay with it.

Anyway, both two way can also solve the problem of driver side, I think we need to get problem fixed first because this is a very basic case.

Reply:
> 

Thanks for your analysis. Driver conFileMaps sets namespace in the [same ways](https://github.com/apache/spark/blob/ddc77fb906cb3ce1567d277c2d0850104c89ac25/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala#L104-L108)

Reply:
@dcoliversun @Yikun thanks for all your comments.

> Driver conFileMaps sets namespace in the same ways

It seems to me that it does not propagate namespace configured by ‘ spark.kubernetes.namespace’ to buildCpnfigMap. We don’t explicitly assign the namespace of configmap before #34983, so it is fine. After we assign the namespace of configmap explicit, it is possible to cause error (on my local) when the true namespace is not propagated.